docs/gallery/general/modifying_data.py

```diff
@@ -0,0 +1,188 @@
+"""
+.. _modifying_data:
+
+Modifying data in an NWB file
```

Comment:
I think the term "modifying" is confusing here. The reason being that folks will typically think about modifying individual fields, i.e., datasets and attributes. I think we should either add a section on how to update datasets and attributes or change the title to something like "Adding/Removing containers from an NWB file".